### ISSUE TITLE
Error when get a document with discriminator array field from DB

### ISSUE THREAD

**Issue by lutherman** [vkarpov15/mongoose-lean-virtuals#39]
Hi,

If get a document with discriminator array field from database with mongoose-lean-virtuals@0.6.8, an error will occur even if don't have any virtual fields. 

Below is the test code to reproduce the problem:
```javascript
const mongoose = require('mongoose');
const mongooseLeanVirtuals = require('mongoose-lean-virtuals');
const ObjectId = mongoose.Types.ObjectId;
const Schema = mongoose.Schema;
mongoose.connect('mongodb://localhost:27017/test', { useNewUrlParser: true, useUnifiedTopology: true });



const testSchema = new Schema({
  title: String,
  values: {
    type: [new Schema({
      fieldId: { type: Schema.Types.ObjectId, required: true },
      value: Schema.Types.Mixed,
    })]
  },
});

testSchema.path('values').discriminator('String', new Schema({
  fieldId: { type: Schema.Types.ObjectId, required: true },
  value: {
    type: String,
  },
}));

testSchema.path('values').discriminator('Multi-String', new Schema({
  fieldId: { type: Schema.Types.ObjectId, required: true },
  value: {
    type: [String],
    required: true
  },
}));

testSchema.plugin(mongooseLeanVirtuals);

const testModel = mongoose.model('test', testSchema);

describe('test', () => {
  before('Clear collection and add sample document', async () => {
    await testModel.deleteMany({});
    await new testModel({
      title: 'test',
      values: [{
        fieldId: new ObjectId(),
        value: 'abc',
        __t: 'String'
      },
      {
        fieldId: new ObjectId(),
        value: [
          'abc',
          'sfadsf',
          'bvbd'
        ],
        __t: 'Multi-String'
      }]
    }).save();
  });
  after('DB disconnect', async () => {
    await mongoose.disconnect();
  });
  it('Can it run without error?', async () => {
    const testDoc = await testModel.findOne({ title: 'test' }).lean({ virtuals: true });
    console.log(JSON.stringify(testDoc));
  });
});
```

And the error message is:
```
$ mocha index.js


  test
    1) Can it run without error?


  0 passing (66ms)
  1 failing

  1) test
       Can it run without error?:
     TypeError: Cannot read property 'Multi-String' of undefined
      at findCorrectDiscriminator (node_modules/mongoose-lean-virtuals/index.js:127:52)
      at Array.find (<anonymous>)
      at attachVirtualsToDoc (node_modules/mongoose-lean-virtuals/index.js:125:40)
      at applyVirtualsToResult (node_modules/mongoose-lean-virtuals/index.js:78:7)
      at model.Query.attachVirtuals (node_modules/mongoose-lean-virtuals/index.js:71:10)
      at applyVirtualsToChildren (node_modules/mongoose-lean-virtuals/index.js:111:20)
      at model.Query.attachVirtuals (node_modules/mongoose-lean-virtuals/index.js:70:3)
      at model.Query.<anonymous> (node_modules/mongoose-lean-virtuals/index.js:43:22)
      at next (node_modules/kareem/index.js:198:31)
      at Kareem.execPost (node_modules/kareem/index.js:217:3)
      at _cb (node_modules/kareem/index.js:307:15)
      at /Users/bit072/repo/mongoose-lean-virtuals-discriminator-error/node_modules/mongoose/lib/query.js:4445:12
      at _completeOneLean (node_modules/mongoose/lib/query.js:3635:10)
      at model.Query.Query._completeOne (node_modules/mongoose/lib/query.js:2101:7)
      at Immediate.<anonymous> (node_modules/mongoose/lib/query.js:2146:10)
      at Immediate.<anonymous> (node_modules/mquery/lib/utils.js:116:16)
      at processImmediate (internal/timers.js:456:21)
```

------------------------------
nodejs: 12.18.0
mongodb: 4.2.7
mongoose: 5.10.1
mongoose-lean-virtuals: 0.6.8
mocha: 8.1.2

**Comment by lutherman** [comment]
@vkarpov15 
Sorry, might i know if the bug has been fixed?
Or, is it a wrong report because isn't a bug?

**Comment by vkarpov15** [comment]
@lutherman yes, the issue has been fixed in v0.6.9 :+1:

**Comment by lutherman** [comment]
@vkarpov15 
Oh, I forgot to check the commits or changelog firstly.
Thank you so much for fixing the problem.
